### PR TITLE
Add initial r5q support

### PR DESCRIPTION
**Pull request by Icesito68**
### Description

The buttons work partially, only the top one is missing, and the ufs is read perfectly, also the uefi interface is perfect

### Checklist

* [y] Have you tested the change you are submitting?
* [y] Is the commits history nice and clean?

### CODE REVIEW

**Comment by ungeskriptet** [comment]
You can use your ButtonsDxe from extracted xbl just fine. SimpleInit is made to work with unpatched ButtonsDxe, the only usecase for patched ButtonsDxe is to have enter working in GRUB. Using drivers from other devices could potentially be dangerous.

**Comment by Icesito68** [comment]
but if I'm not mistaken I have not used drivers from another device for the buttons, unless I was wrong

**Comment by ungeskriptet** [comment]
> but if I'm not mistaken I have not used drivers from another device for the buttons, unless I was wrong

You are, look in file Platform/Samsung/sm8150/r5q.fdf.inc line 11.

File: Platform/Samsung/sm8150/r5q.fdf.inc [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+FILE DRIVER = 8e9bd160-b184-11df-94e2-0800200c9a66 {
+    SECTION DXE_DEPEX = Platform/EFI_Binaries/Drivers/sm8150/DALSys/DALSys.depex
+    SECTION PE32 = Platform/EFI_Binaries/Drivers/Devices/vayu/DALSys/DALSys.efi
+    SECTION UI = "DALSys"
+}
+
+FILE DRIVER = 5bd181db-0487-4f1a-ae73-820e165611b3 {
+    SECTION DXE_DEPEX = Platform/EFI_Binaries/Drivers/sm8150/ButtonsDxe/ButtonsDxe.depex
+   SECTION PE32 = Platform/EFI_Binaries/Drivers/Devices/vayu/ButtonsDxe/ButtonsDxe.efi
```
Comment (qaz6750) [inline Platform/Samsung/sm8150/r5q.fdf.inc:11]: I don't understand why we should use ButtonsDxe and DALSys of other devices. Even though they can work, I don't think they should be used in this way

**Review by qaz6750 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**